pages/add-clinic.vue

```diff
@@ -124,29 +124,61 @@ export default {
     prefectureList: json.prefectures,
     prefecture: "",
     city: "",
-    cityRules: [
-      (v) => !!v || "City is required",
-      (v) => (v && v.length >= 2) || "City name must be at least 2 characters",
-    ],
     ward: "",
-    wardRules: [
-      (v) => !!v || "Ward is required",
-      (v) => (v && v.length >= 2) || "Ward name must be at least 2 characters",
-    ],
     name: "",
-    nameRules: [
-      (v) => !!v || "Clinic name is required",
-      (v) =>
-        (v && v.length >= 2) || "Clinic name must be at least 2 characters",
-    ],
     note: "",
     website: "",
-    websiteRules: [
-      (v) => !!v || "Website URL is required",
-      (v) => (v && v.length >= 5) || "Please enter a valid URL",
-    ],
   }),
 
+  computed: {
+    cityRules() {
+      const minLength = 2;
+
+      return [
+        (v) => !!v || this.$i18n.t("add-clinic.validations.cityRequired"),
+        (v) =>
+          (v && v.length >= minLength) ||
+          this.$i18n.t("add-clinic.validations.cityValidation", [minLength]),
+      ];
+    },
+
+    wardRules() {
+      const minLength = 2;
+
+      return [
+        (v) => !!v || this.$i18n.t("add-clinic.validations.wardRequired"),
+        (v) =>
+          (v && v.length >= minLength) ||
+          this.$i18n.t("add-clinic.validations.wardValidation", [minLength]),
+      ];
+    },
+
+    nameRules() {
+      const minLength = 2;
+
+      return [
+        (v) => !!v || this.$i18n.t("add-clinic.validations.clinicRequired"),
+        (v) =>
+          (v && v.length >= minLength) ||
+          this.$i18n.t("add-clinic.validations.clinicValidation", [minLength]),
+      ];
+    },
+
+    websiteRules() {
+      // Simple URL pattern check: it must starts with either "http" or
+      // "https" and contains one character after the prefix.
+      // Regex can be tested on https://regex101.com/
+      const urlPattern = /^http[s]?:\/\/(.+)/;
+
+      return [
+        (v) => !!v || this.$i18n.t("add-clinic.validations.websiteRequired"),
+        (v) =>
+          urlPattern.test(v) ||
+          this.$i18n.t("add-clinic.validations.websiteValidation"),
+      ];
+    },
```

Comment:
Not really an issue, but knowing this is a form that any person can contribute adding a clinic, maybe the URL validation will prevent contributors with less technological knowledge from helping when their `clinic-site.jp` doesn't validate because it doesn't start with `http(s)://`. If the volume of contributions isn't that high, I would think it would be better to leave this more open and transfer the work of checking to the moderators.
What do you think?

Reply:
True that it could be left to moderators. But it must be an openable link (For being used here https://github.com/ourjapanlife/findadoc-frontend/blob/main/components/CancellationList.vue#L59). So at worst, it should be some `www.(something)`. Otherwise we can add a link the review page (that I haven't checked) and if the link opens, it is OK

Reply:
I think this point can be closed: I will use the rule introduced in #169

Reply:
Yes, we would like to use the Vuelidate URL rule, and we are getting a lot of junk here after the recent publicity, so we want to be strict on this field.

Reply:
Yea I only saw yesterday in the chat. I had no idea that amount of junk was being sent.